docs/DeveloperGuide.md

```diff
@@ -1,34 +1,255 @@
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+## 2. Design 
+### 2.1. Architecture 
+### 2.2. UI Component 
+### 2.3. Parser Component 
+### 2.4. Command Classes 
+### 2.5. Storage Component 
 
-## User Stories
+## 3. Implementation
 
-|Version| As a ... | I want to ... | So that I can ...|
+{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.} 
+
+### 3.1. Save feature
+#### Current Implementation
+The save mechanism is facilitated by `AliasStorage`, `HistoryRouteStorage`, `NotesStorage`, `DailyRouteStorage`, `FavouriteLocationsStorage` subclasses. </br>
+They extend `Storage` (superclass) with a feature to save the block aliases, history of visited routes, tagged notes, daily routes and favourite locations, stored internally as a `aliasList`,  `historyList`, `notesList`, `dailyRouteList` and `favouritesList`. <br />
+Additionally, they implement the following operations: <br/>
+- `AliasStorage#overwriteAliasListFile()` —  Saves all aliases given by user to blocks into `aliasList`. <br />
+- `AliasStorage#loadAlias()`   —  Restores all aliases given by user to blocks from `aliasList`. <br />
+- `HistoryRouteStorage#overwriteHistoryListFile()` —  Saves the current list of the 10 most recently visited routes in its history into `historyList`. <br />
+- `HistoryRouteStorage#loadHistory()` —  Restores the previous list of the 10 most recently visited routes in its history from `historyList`. <br />
+- `NotesStorage#overwriteNotesListFile()` —  Saves all notes tagged to a location into `notesList`. <br />
+- `NotesStorage#loadNotes()` —  Restores all notes tagged to a location from `notesList`. <br />
+- `DailyRouteStorage#loadDailyRoute()` —  Saves all the daily routes that user wants to see for each day of the week into `dailyRouteList`. <br />
+- `DailyRouteStorage#overwriteDailyRouteFile()` Restores all the daily routes that user wants to see from `dailyRouteList`. <br />
+- `FavouriteLocationsStorage#overwriteFavouritesListFile()` —  Saves the current list of all the locations that the users are interested in keeping in `favouritesList`. <br />
+- `FavouriteLocationsStorage#loadFavourites()` —  Restores the previous list of the all the locations that the users are interested in keeping from `favouritesList`. <br />
+
+These operations are exposed in the `Storage` class  as `Storage#loadAlias()`, `Storage#overwriteAliasListFile()`, `Storage#loadHistory()`, `Storage#overwriteHistoryListFile()` , `Storage#loadNotes()`, `Storage#overwriteNotesListFile()`, `Storage#loadDailyRoute()`, `Storage#overwriteDailyRouteFile()` , `Storage#loadFavourites()` and `Storage#overwriteFavouritesListFile()`. <br />
+The image below shows an overview for the storage component, which consist of Storage class and its four subclasses.
+![img.png](Overview%20for%20Safe%20Feature.png)
```

Comment:
maybe this is a non standard UML notation for the diagram?

Reply:
![image](https://user-images.githubusercontent.com/59407923/113076235-2cd96780-9201-11eb-9570-88cf1ea870f6.png)

Reply:
Maybe you can check https://github.com/nus-cs2113-AY2021S2/forum/issues/54